### ISSUE TITLE
task02: CNN for cats and dogs

### ISSUE THREAD

**Issue by Revilllo7** [Revilllo7/computational-intelligence-class#62]
a) get familiar with a given tutorial
https://docs.pytorch.org/tutorials/beginner/transfer_learning_tutorial.html

b) load the dataset and prepare it
- unpacking and loading the images
- label creation `dog` or `cat` based on file name
- preparing the training, validation and testing dataset
- transforming the images in PyTorch (example: size adjustments, tensor conversion, normalisation and data augmentation)

c) build and train the model on the training dataset.
Architecture can be standard (example: couple layers Conv2d, ReLu, MaxPool2d) and then fully connected layers. Implementation should be done in PyTorch.

d) validate the model
show the learning curves for the training and validation dataset like loss function across epochs.

Questions:
- Are there any cats resembling dogs or vice verse in the dataset?
- How many such cases can be detected?
- Can you find and show the specific images categorised by the model as false positives and false negative?

e) re-run the experiment from subpoints c-d for different network config, example:
- SGD, Adam optimizer
- Activation function
- Layer count
- Filter count
- Dropout utilisation
- data augmentation parameters

Report which config worked the best and showcase it's results, especially:
- learning curves
- final accuracy
- confusion matrix

In case of long training, remember that you can always save the model and continue the training later.

f) Question: can you use a pre-trained model and then train it for better recognition between cats and dogs?
run an experiment with transfer learning in PyTorch. For example with a model such as `ResNes`, `MobileNet` or `EfficientNet` and compare the results with a model trained from 0.

The official PyTorch tutorial shows such work scheme for object classification
